_blog/2023-12-29-using-loco-rust-rails.mdx

```diff
@@ -0,0 +1,316 @@
+In this article we're going to talk about how you can get started with Loco - a new web framework that builds on Axum. We will cover getting started using controllers, migrations, middleware, static files and handling a database.
+
+## Getting Started
+
+To get started, you will need to make sure to have Loco's CLI installed by using the following:
+
+```bash
+cargo install loco-cli
+```
+
+Loco also uses `sea-orm-cli` to carry out database migrations. You can install it using the following:
+
+```bash
+cargo install sea-orm-cli
+```
+
+Now we've installed all of the required packages, we can get initialise our project. We can get started by using `loco new` and then selecting the SaaS application which will give us an app with full functionality. The given name we will be using for the app will be `example-app`. Don't forget to `cd` into the project folder!
+
+When you're writing your API, you will probably want to spin up a local Docker database for database testing. In that case, you will wnat to use this docker command:
+```bash
+$ docker run -d -p 5432:5432 -e POSTGRES_USER=loco -e POSTGRES_DB=example-app_development -e POSTGRES_PASSWORD="loco" postgres:15.3-alpine
+```
+
+## Routing in Loco
+
+To get started, we will generate a "scaffold". This generates a controller, model and migration all at the same time. We can also add pre-generated field names and types beforehand, which you can find more about [here.](https://loco.rs/docs/the-app/models/#migrations) We can create our own scaffold below:
+
+```bash
+cargo loco generate scaffold item name:string! description:string quantity:int!
+```
+
+This will generate a controller, model and migration for a table named `my-scaffold` with:
+- A non-nullable name field
+- A nullable description field
+- A non-nullable quantity field
+
+Once the scaffold is done, you will need to run `cargo loco db migrate`. This generates the migrations and entities required for your app. You will notice also that your controller and other relevant parts have been added in `app.rs`, so there is no need to do anything regarding this.
```

Comment:
In general when generating a scaffold it will generate entities and run migrations automatically in development